awx/ui_next/src/components/Schedule/shared/ScheduleForm.jsx

```diff
@@ -0,0 +1,230 @@
+        isRequired
+        isValid={!startDateTimeMeta.touched || !startDateTimeMeta.error}
+        label={i18n._(t`Start date/time`)}
+      >
+        <input
+          className="pf-c-form-control"
+          type="datetime-local"
+          id="schedule-start-datetime"
+          step="1"
+          {...startDateTime}
+        />
+      </FormGroup>
+      <FormGroup
+        name="timezone"
+        fieldId="schedule-timezone"
+        helperTextInvalid={timezoneMeta.error}
+        isRequired
+        isValid={!timezoneMeta.touched || !timezoneMeta.error}
+        label={i18n._(t`Local time zone`)}
+      >
+        <AnsibleSelect
+          id="schedule-timezone"
+          data={zoneOptions}
+          {...timezone}
+        />
+      </FormGroup>
+      <FormGroup
+        name="frequency"
+        fieldId="schedule-requency"
+        helperTextInvalid={frequencyMeta.error}
+        isRequired
+        isValid={!frequencyMeta.touched || !frequencyMeta.error}
+        label={i18n._(t`Run frequency`)}
+      >
+        <AnsibleSelect
+          id="schedule-frequency"
+          data={[
+            { value: 'none', key: 'none', label: i18n._(t`None (run once)`) },
+            { value: 'minute', key: 'minute', label: i18n._(t`Minute`) },
+            { value: 'hour', key: 'hour', label: i18n._(t`Hour`) },
+            { value: 'day', key: 'day', label: i18n._(t`Day`) },
+            { value: 'week', key: 'week', label: i18n._(t`Week`) },
+            { value: 'month', key: 'month', label: i18n._(t`Month`) },
+            { value: 'year', key: 'year', label: i18n._(t`Year`) },
+          ]}
+          {...frequency}
+        />
+      </FormGroup>
+      {frequency.value !== 'none' && (
+        <SubFormLayout>
+          <Title size="md">{i18n._(t`Frequency Details`)}</Title>
+          <FormColumnLayout>
+            <FrequencyDetailSubform />
+          </FormColumnLayout>
+        </SubFormLayout>
+      )}
+    </>
+  );
+}
+
+function ScheduleForm({
+  handleCancel,
+  handleSubmit,
+  i18n,
+  schedule,
+  submitError,
+  ...rest
+}) {
+  const {
+    request: loadZoneInfo,
+    error: contentError,
+    contentLoading,
+    result: zoneOptions,
+  } = useRequest(
+    useCallback(async () => {
+      const { data } = await SchedulesAPI.readZoneInfo();
+      return data.map(zone => {
+        return {
+          value: zone.name,
+          key: zone.name,
+          label: zone.name,
+        };
+      });
+    }, [])
+  );
+
+  useEffect(() => {
+    loadZoneInfo();
+  }, [loadZoneInfo]);
+
+  if (contentError) {
+    return <ContentError error={contentError} />;
+  }
+
+  if (contentLoading) {
+    return <ContentLoading />;
+  }
+
+  return (
+    <Config>
+      {() => {
+        const now = new Date();
+        const closestQuarterHour = new Date(
+          Math.ceil(now.getTime() / 900000) * 900000
+        );
+        const tomorrow = new Date(closestQuarterHour);
+        tomorrow.setDate(tomorrow.getDate() + 1);
```

Comment:
What does this do &  and do we need it in the return object?

Reply:
Yea I'll move it out of the return object.  This logic is used to set the initial start date to the nearest quarter hour (and then 1 day past that for the end date) but it doesn't need to be in the return.

Reply:
Moved here - https://github.com/ansible/awx/pull/6419/commits/693d8da14c37aa146212deb128f15ceed697c117